File: test-suite-caffeine-java/src/test/java/io/micronaut/cache/MyRemovalHandler.java
package io.micronaut.cache;

import com.github.benmanes.caffeine.cache.RemovalCause;
import jakarta.inject.Singleton;

import java.util.ArrayList;
import java.util.List;

@Singleton
public class MyRemovalHandler {

    private List<String> removals = new ArrayList<>();

    void handle(String key, Integer value, RemovalCause cause) {
        removals.add(key + "|" + value + "|" + cause);
    }

    public List<String> getRemovals() {
        return removals;
    }
}

File: test-suite-caffeine-kotlin/src/test/kotlin/io/micronaut/cache/MyRemovalHandler.kt
package io.micronaut.cache

import com.github.benmanes.caffeine.cache.RemovalCause
import jakarta.inject.Singleton

@Singleton
class MyRemovalHandler {

    val removals: MutableList<String> = ArrayList()

    fun handle(key: String, value: Int, cause: RemovalCause) {
        removals.add("$key|$value|$cause")
    }
}
